staging/src/k8s.io/client-go/rest/urlbackoff.go

```diff
@@ -49,22 +63,38 @@ type URLBackoff struct {
 type NoBackoff struct {
 }
 
-func (n *NoBackoff) UpdateBackoff(actualUrl *url.URL, err error, responseCode int) {
+func (n *NoBackoff) UpdateBackoff(actualURL *url.URL, err error, responseCode int) {
 	// do nothing.
 }
 
-func (n *NoBackoff) CalculateBackoff(actualUrl *url.URL) time.Duration {
+func (n *NoBackoff) UpdateBackoffWithContext(ctx context.Context, actualURL *url.URL, err error, responseCode int) {
+	// do nothing.
+}
+
+func (n *NoBackoff) CalculateBackoff(actualURL *url.URL) time.Duration {
+	return 0 * time.Second
+}
+
+func (n *NoBackoff) CalculateBackoffWithContext(ctx context.Context, actualURL *url.URL) time.Duration {
 	return 0 * time.Second
 }
 
 func (n *NoBackoff) Sleep(d time.Duration) {
 	time.Sleep(d)
 }
 
+func (n *NoBackoff) SleepWithContext(ctx context.Context, d time.Duration) {
```

Comment:
This is one reason why unit tests were slower.

https://github.com/kubernetes/kubernetes/blob/4e4ea1bad148cb1b0530e923c88def4a733b28fc/staging/src/k8s.io/client-go/rest/request_test.go#L2607-L2613

inherited `NoBackoff.SleepWithContext` without overriding the `SleepWithContext` method.

One could argue that this is an API break, but normally adding methods to a struct is not considered an API break and it's not flagged by apidiff as one. I think code which is meant to implement an interface should do that itself and delegate to someone else where needed, so I think the test code was at fault here. I've fixed it and most tests run quickly again.

The remaining problem is `TestBackoffLifecycle` which depends on a fake clock. The new code now uses `Clock.NewTimer` instead of `Clock.Sleep` and something there doesn't seem to fake time properly. I'm still looking.